src/distance.jl

```diff
@@ -1,45 +1,52 @@
 """
-distance(tree::Tree,
-         node1::Int64,
-         node2::Int64)
+    distance(tree::Tree, node1, node2)
 
 Distance between two `Node`s on a `Tree`
 """
-function distance(tree::Tree,
-                  node1::Int64,
-                  node2::Int64)
-  path = branchpath(tree, node1, node2)
-  dist = 0.
-  for i in path
-    dist += tree.branches[i].length
-  end
-  return dist
+function distance(tree::AbstractTree, node1, node2)
+    path = branchpath(tree, node1, node2)
+    dist = 0.0
+    for b in path
+        dist += getlength(tree, b)
+    end
+    return dist
 end
 
 
 """
-distance(tree::Tree)
+    distance(tree::Tree)
 
 Pairwise distances between all leaf `Node`s on a `Tree`
 """
-function distance(tree::Tree)
-  leaves = findleaves(tree)
-  return [distance(tree, i, j) for i in leaves, j in leaves]
+function distance(tree::AbstractTree)
+    leaves = findleaves(tree)
+    return [distance(tree, i, j) for i in leaves, j in leaves]
 end
 
+"""
+    distance(tree::Tree, nodes::AbstractVector)
 
+Distance between a `Node` and it's associated root
 """
-distance(tree::Tree,
-         node::Int64)
+function distance(tree::AbstractTree, nodes::AbstractVector)
+    return map(node -> distance(tree, node), nodes)
+end
+
+"""
+    distance(tree::Tree, node)
 
 Distance between a `Node` and it's associated root
 """
-function distance(tree::Tree,
-                  node::Int64)
-  path = branchpath(tree, node)
-  dist = 0.
-  for i in path
-    dist += tree.branches[i].length
-  end
-  return dist
+function distance(tree::AbstractTree, node)
+    path = branchpath(tree, node)
+    dist = 0.0
+    for b in path
+        dist += getlength(tree, b)
+    end
+    return dist
+end
+
+function getrootdistance(tree::AbstractTree, label)
```

Comment:
No, and to be honest, I only realised that (and included it in this file rather than where it had been originally) after I realised that. However, I think that `distance(tree, node)` is not intuitive for answering that question, so I would prefer this function rather than `distance()` to calculate it.

PS My `...height...` functions are intended to calculate *absolute* heights rather than _relative_ distances...